stdlib/universe/map.go

```diff
@@ -97,159 +105,264 @@ func createMapTransformation(id execute.DatasetID, mode execute.AccumulationMode
 		return nil, nil, errors.Newf(codes.Internal, "invalid spec type %T", spec)
 	}
 
-	if feature.VectorizedMap().Enabled(a.Context()) {
-		return newMapTransformation2(a.Context(), id, s, a.Allocator())
-	}
-
-	cache := execute.NewTableBuilderCache(a.Allocator())
-	d := execute.NewDataset(id, mode, cache)
-	t, err := NewMapTransformation(a.Context(), s, d, cache)
+	return newMapTransformation(a.Context(), id, s, a.Allocator())
+}
 
-	if err != nil {
-		return nil, nil, err
+func newMapTransformation(ctx context.Context, id execute.DatasetID, spec *MapProcedureSpec, mem memory.Allocator) (execute.Transformation, execute.Dataset, error) {
+	tr := &mapTransformation{
+		ctx: ctx,
+		fn: &mapRowFunc{
+			fn: execute.NewRowMapFn(
+				spec.Fn.Fn,
+				compiler.ToScope(spec.Fn.Scope),
+			),
+		},
 	}
-	return t, d, nil
+	return execute.NewGroupTransformation(id, tr, mem)
 }
 
 type mapTransformation struct {
-	execute.ExecutionNode
-	d        execute.Dataset
-	cache    execute.TableBuilderCache
-	ctx      context.Context
-	fn       *execute.RowMapFn
-	mergeKey bool
+	ctx context.Context
+	fn  mapFunc
 }
 
-func NewMapTransformation(ctx context.Context, spec *MapProcedureSpec, d execute.Dataset, cache execute.TableBuilderCache) (*mapTransformation, error) {
-	fn := execute.NewRowMapFn(spec.Fn.Fn, compiler.ToScope(spec.Fn.Scope))
-	return &mapTransformation{
-		d:        d,
-		cache:    cache,
-		fn:       fn,
-		ctx:      ctx,
-		mergeKey: spec.MergeKey,
-	}, nil
-}
+func (m *mapTransformation) Process(
```

Comment:
This looks like a big change, but all of this code here was copied from `map2.go`. I did not need to alter it.

Reply:
Nit: If you delete the previous file first, then do `git mv` you can probably preserve the history better (might need to delete in one commit, move in the next).

Reply:
Ah, I will definitely do that next time.